pkg/upgrader/broker/v0.15.0/upgrader.go

```diff
@@ -0,0 +1,182 @@
+			continue
+		}
+
+		// Ok, there are differences, apply the patch
+		logger.Infof("Patching Broker \"%s/%s\" with %q", broker.Namespace, broker.Name, string(patch))
+		patched, err := brokerClient.Patch(broker.Name, types.MergePatchType, patch)
+		if err != nil {
+			logger.Warnf("Failed to patch \"%s/%s\" : %v", broker.Namespace, broker.Name, err)
+			return err
+		}
+		logger.Infof("Patched \"%s/%s\" successfully new Spec: %+v", broker.Namespace, broker.Name, patched.Spec)
+	}
+	return nil
+}
+
+// Process a broker, create a ConfigMap representing the ChannelTemplate
+// and point the Config to it.
+// Returns non-empty patch bytes if a patch is necessary.
+func processBroker(ctx context.Context, broker v1alpha1.Broker) ([]byte, error) {
+	logger := logging.FromContext(ctx)
+	if broker.Spec.ChannelTemplate == nil || broker.Spec.ChannelTemplate.Kind == "" {
+		logger.Infof("Broker \"%s/%s\" is not using channeltemplate, skipping...", broker.Namespace, broker.Name)
+		return []byte{}, nil
+	}
+
+	modified := broker.DeepCopy()
+
+	// If the Broker already has a Config, don't modify it, just nil out the channel template.
+	if broker.Spec.Config == nil || broker.Spec.Config.Name == "" {
+		cm, err := createConfigMap(ctx, broker)
+		if err != nil {
+			return []byte{}, err
+		}
+		modified.Spec.Config = cm
+	}
+	modified.Spec.ChannelTemplate = nil
+
+	patch, err := duck.CreateMergePatch(broker, modified)
+	if err != nil {
+		logger.Warnf("Failed to create patch for \"%s/%s\" : %v", broker.Namespace, broker.Name, err)
+		return []byte{}, err
+	}
+	logger.Infof("Patch for \"%s/%s\": %q", broker.Namespace, broker.Name, string(patch))
+	// If there is nothing to patch, we are good, just return.
+	// Empty patch is {}, hence we check for that.
+	if len(patch) <= 2 {
+		return []byte{}, nil
+	}
+	return patch, nil
+}
+
+func createConfigMap(ctx context.Context, broker v1alpha1.Broker) (*duckv1.KReference, error) {
+	// Generating the spec portion is a bit goofy cause we have turn the runtime raw into
+	// a yaml blob that we stick into the configmap (with proper indentation).
+	data := ""
+	if broker.Spec.ChannelTemplate.Spec != nil {
+		bytes, err := yaml.JSONToYAML(broker.Spec.ChannelTemplate.Spec.Raw)
+		if err != nil {
+			return nil, err
+		}
+		logging.FromContext(ctx).Infof("BYTES: %q", string(bytes))
+		data = fmt.Sprintf(`
+  apiVersion: %q
+  kind: %q
+  spec:
+    %s
+`, broker.Spec.ChannelTemplate.APIVersion, broker.Spec.ChannelTemplate.Kind, strings.ReplaceAll(strings.TrimSpace(string(bytes)), "\n", "\n    "))
+	} else {
+		data = fmt.Sprintf(`
+  apiVersion: %q
+  kind: %q
+`, broker.Spec.ChannelTemplate.APIVersion, broker.Spec.ChannelTemplate.Kind)
+	}
+
+	cm := &corev1.ConfigMap{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "ConfigMap",
+			APIVersion: "v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "broker-auto-gen-config-" + broker.Name,
```

Comment:
is `broker-auto-gen-config` really correct here ? 

It is auto generated, but for upgraade reasons ... :thinking: 

`broker-auto-upgrate gen-config` ? 

sorry - just thinking here out loud - Pleaes ignore if you disagree, which is perfectly fine :-)